tests/Kconfig

```diff
@@ -23,20 +23,49 @@ config TEST_SHELL
 
 config TEST_EXTRA_STACKSIZE
 	int "Test function extra thread stack size"
+	default 1024 if COVERAGE_GCOV
 	default 768 if XTENSA
 	default 0
 	help
 	  Additional stack for tests on some platform where default is not
 	  enough.
 
+config HAS_COVERAGE_SUPPORT
+	bool
+	help
+	  The code coverage report generation is only available on boards
+	  with enough RAM.
+
+config PREVENT_COVERAGE
```

Comment:
Please get rid of that, you should be able to just use CONFIG_COVERAGE=n

Reply:
this is a problem with how we merge the kconfigs in our build system.
even if I disable this kconfig. A new .conf file will be applied on top of this via cmake and sanitycheck. which will enable the COVERAGE even if i have said no previously.